crates/transaction-pool/src/lib.rs

```diff
@@ -379,17 +391,16 @@ where
             .collect()
     }
 
-    /// Validates the given transaction
-    async fn validate(
+    /// Validates all transactions with their individual origins.
+    ///
+    /// This returns the validated transactions in the same order as input.
+    async fn validate_all_with_origins(
         &self,
-        origin: TransactionOrigin,
-        transaction: V::Transaction,
-    ) -> (TxHash, TransactionValidationOutcome<V::Transaction>) {
-        let hash = *transaction.hash();
-
-        let outcome = self.pool.validator().validate_transaction(origin, transaction).await;
-
-        (hash, outcome)
+        transactions: Vec<(TransactionOrigin, V::Transaction)>,
+    ) -> Vec<(TransactionOrigin, TransactionValidationOutcome<V::Transaction>)> {
+        let origins: Vec<_> = transactions.iter().map(|(origin, _)| *origin).collect();
```

Comment:
I see,

hmm, we could perhaps consider making this part of the response, but this isn't too bad because this usecase isn't that common